xapian-core/cluster/clusterinternal.h

```diff
@@ -0,0 +1,120 @@
+/** @file clusterinternal.h
+ *  @brief Cluster API
+ */
+/* Copyright (C) 2017 Richhiey Thomas
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+
+#include <xapian/intrusive_ptr.h>
+
+#include <unordered_map>
+#include <vector>
+
+using namespace std;
+
+/** Internal class for ClusterSet
+ */
+class Xapian::ClusterSet::Internal : public Xapian::Internal::intrusive_base {
+  private:
+    /// Copies are not allowed
+    Internal(const Internal &);
+
+    /// Assignment is not allowed
+    void operator=(const Internal &);
+
+   /** A vector storing the clusters that are created by the
+    *  clusterers
+    */
+    vector<Cluster> clusters;
+  public:
```

Comment:
The whole comment is indented by one column too few (the `/` should line up with the `v` in `vector`).